packages/astro/src/content/loaders/types.ts

```diff
@@ -49,14 +49,27 @@ export interface LoaderContext {
 	entryTypes: Map<string, ContentEntryType>;
 }
 
-export interface Loader {
+export type Loader<T extends ZodSchema = never> = {
 	/** Unique name of the loader, e.g. the npm package name */
 	name: string;
 	/** Do the actual loading of the data */
 	load: (context: LoaderContext) => Promise<void>;
-	/** Optionally, define the schema of the data. Will be overridden by user-defined schema */
-	schema?: ZodSchema | Promise<ZodSchema> | (() => ZodSchema | Promise<ZodSchema>);
-}
+} & ([T] extends [never]
+	?
+			| {
+					/** Optionally, define the schema of the data. Will be overridden by user-defined schema */
+					schema?: T;
+			  }
+			| {
+					getSchemaContext?: () => Promise<{
+						schema: ZodSchema;
```

Comment:
```suggestion
						schema: T;
```
Should this be `T`?

Reply:
No because we don't infer the type from this, it's only used at runtime